### ISSUE TITLE
Gpsprune fails to start after java was installed

### ISSUE THREAD

**Issue by Alekssavin** [activityworkshop/GpsPrune#82]
Previously I had no Java installed on my `Opensuse 15.4` distribution and was not able to use Three d view in `GpsPrune`. Now I installed it and I am no longer able to run `GpsPrune`.
```
user@localhost:~> gpsprune
Unrecognized option: --add-opens
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.
```

[This](https://www.ibm.com/support/pages/build-fails-unrecognized-option-add-opens) hints that it may be a java version issue. I have `java 1.8.0_381` installed, which is the latest I could find on the Oracle website. What should I do to make `GpsPrune` work again?
```
user@localhost:~> java -version
java version "1.8.0_381"
Java(TM) SE Runtime Environment (build 1.8.0_381-b09)
Java HotSpot(TM) 64-Bit Server VM (build 25.381-b09, mixed mode)
```

**Comment by activityworkshop** [comment]
> Gpsprune fails to start after java was installed

> Previously I had no Java installed on my Opensuse 15.4 distribution

I think there may be some confusion between "java" being installed, and "java3d" being installed.  These are two separate things.

If you do not have java installed, then GpsPrune will not run at all.  If you have java installed but do not have the additional java3d installed, then you will be able to run GpsPrune, but not the Three-D view.  So I think this is what you previously had.

> Unrecognized option: --add-opens

> I have java 1.8.0_381 installed, which is the latest I could find on the Oracle website.

Correct - as your link describes, you cannot use the "--add-opens" command line parameters with java 8, as these are not recognised.  So whatever your start script or shortcut is behind the command "gpsprune", this needs to lose the "--add-opens" parameters.

Question 1: where did you get GpsPrune from?  Did you install it from a package (including the executable `/usr/bin/gpsprune` )?  Or did you get the jar separately from somewhere else?

If you're using a package, this may not be compatible with java 8, so you might need to upgrade your java to either 11 or 14.  You can probably do this with an OpenSuse OpenJdk package called something like `java-11-openjdk` - this is probably more reliable than downloading things from an Oracle website.

Option 1: Keep your Java 8, use a Java 8-compatible jar of GpsPrune, and launch it yourself with a `java` command
Option 2: Upgrade your Java to OpenJdk11, then your existing `gpsprune` launch script may (or may not) work.

Getting the java3d part to work is a separate problem, but you should be able to start GpsPrune without java3d as a first step.

**Comment by Alekssavin** [comment]
I have `java-17-openjdk` as well as `java-11-openjdk` already installed, so the Option 2 doesn't help.

I previously had no problem starting the app before I downloaded java from [here](https://www.java.com/en/download/linux_manual.jsp) and run `rpm -ivh jre-8u381-linux-x64.rpm`.

I suppose I've got to dispose of this package I've installed.
```
user@localhost:~> rpm -e jre-1.8-1.8.0_381-9.x86_64
update-alternatives: warning: alternative /usr/lib/jvm/jre-1.8-oracle-x64/bin/java (part of link group java) doesn't exist; removing from list of alternatives
update-alternatives: warning: /etc/alternatives/java is dangling; it will be updated with best choice
update-alternatives: using /usr/lib64/jvm/jre-17-openjdk/bin/java to provide /usr/bin/java (java) in auto mode
```
 Now `gpsprune` is able to run again!

I do have `java-3d` in my packages list, so I'll install it.

Now the Three d mode works!

I seem to have troubles plotting the terrain in this mode, it's a different problem I'll try to figure it out myself or I'll raise another issue.

Thank you very much for your answer, it helped a lot!
